src/Compiler/CompilerOpenFPGA.cpp

```diff
@@ -502,6 +502,14 @@ bool CompilerOpenFPGA::RegisterCommands(TclInterpreter* interp,
   return true;
 }
 
+bool CompilerOpenFPGA::VerifyTargetDevice() const {
+  const bool target = Compiler::VerifyTargetDevice();
+  const bool archFile = FileUtils::FileExists(m_architectureFile);
+  const bool openFpgaArchFile =
+      FileUtils::FileExists(m_OpenFpgaArchitectureFile);
+  return target || (archFile && openFpgaArchFile);
```

Comment:
The test should be return (target || archFile);    for all the stages except bitstream stage.

The openFpgaArchFile is only needed for the bitstream generation stage.

Reply:
Ready